### PR TITLE
Add extra linting

### LINKED ISSUE #4974: Rust 2024 breaking changes
### Vaultwarden Support String

N/A

### Vaultwarden Build Version

N/A

### Deployment method

Build from source

### Custom deployment method

_No response_

### Reverse Proxy

N/A

### Host/Server Operating System

Other

### Operating System Version

_No response_

### Clients

Web Vault

### Client Version

_No response_

### Steps To Reproduce

Add `rust-2024-compatibility` to the list of `deny` lints and compile code. One error is caused by some of the macro definitions. For example, `crate::error::make_error` doesn't compile due to "the `expr` fragment specifier will accept more expressions in the 2024 edition".

### Expected Result

Compilation to succeed.

### Actual Result

Compilation failure.

### Logs

_No response_

### Screenshots or Videos

_No response_

### Additional Context

See [the Migration Guide](https://doc.rust-lang.org/nightly/edition-guide/rust-2024/macro-fragment-specifiers.html) for how to fix this.

### PR DESCRIPTION
**Pull request by BlackDex**
Added extra linting for some code styles.
Also added some of the Rust Edition 2024 lints since our MSRV does not yet support the group lint.
Once we reach 1.81.0 as MSRV we can uncomment it and use that lint.

Closes #4974

### CODE REVIEW

**Comment by BlackDex** [comment]
Hmm, looks like our MSRV doesn't allow those lints yet.

**Comment by BlackDex** [comment]
Only available since v1.81.0 it looks like.

**Comment by BlackDex** [comment]
i Could comment it out for now and enable it once we are there, and leave the rest as-is.

File: src/api/core/accounts.rs [hunk 1]
```diff
@@ -223,7 +223,7 @@ pub async fn _register(data: Json<RegisterData>, mut conn: DbConn) -> JsonResult
         }
 
         if verified_by_invite && is_email_2fa_required(data.organization_user_id, &mut conn).await {
-            let _ = email::activate_email_2fa(&user, &mut conn).await;
+            drop(email::activate_email_2fa(&user, &mut conn).await);
```
Comment (dani-garcia) [inline src/api/core/accounts.rs:226]: Hmm I don't really like the use of drop here, I think this lint is a bit misguided for this use case.

The lint seems to want to make the calling of `Drop` more explicit, but that's not why we're using `let _ =` here, we're using it to indicate that we don't care about the returned `Result` type.

We could also switch to using `.ok()` instead, which I think we also use in other places.

**Review by dani-garcia — COMMENTED** [review]
I think we can comment that `edition_2024_expr_fragment_specifier` out until we bump the MSRV in two months yeah.

File: src/api/core/accounts.rs (hunk 1 above)
Comment (BlackDex) [reply src/api/core/accounts.rs:226]: Yea, if we don't want to traverse the error, the  `.ok()` is good. I don't like `let _` for these cases either, and `.ok()` is more clear then on what we want.

**Review by dani-garcia — APPROVED** [review]

**Pull request merged**